plugins/woocommerce-blocks/bin/webpack-configs.js

```diff
@@ -986,6 +978,151 @@ const getInteractivityAPIConfig = ( options = {} ) => {
+		? {
+				alias,
+				plugins: resolvePlugins,
+		  }
+		: {
+				plugins: resolvePlugins,
+		  };
+	return {
+		entry: getEntryConfig(
+			'cartAndCheckoutFrontend',
+			options.exclude || []
+		),
+		output: {
+			devtoolNamespace: 'wc',
+			path: path.resolve( __dirname, '../build/' ),
+			// This is a cache busting mechanism which ensures that the script is loaded via the browser with a ?ver=hash
+			// string. The hash is based on the built file contents.
+			// @see https://github.com/webpack/webpack/issues/2329
+			// Using the ?ver string is needed here so the filename does not change between builds. The WordPress
+			// i18n system relies on the hash of the filename, so changing that frequently would result in broken
+			// translations which we must avoid.
+			// @see https://github.com/Automattic/jetpack/pull/20926
+			chunkFilename: `[name]-frontend${ fileSuffix }.js?ver=[contenthash]`,
+			filename: ( pathData ) => {
+				// blocksCheckout and blocksComponents were moved from core bundle,
+				// retain their filenames to avoid breaking translations.
+				if (
+					pathData.chunk.name === 'blocksCheckout' ||
+					pathData.chunk.name === 'blocksComponents'
+				) {
+					return `${ paramCase(
+						pathData.chunk.name
+					) }${ fileSuffix }.js`;
+				}
+
+				return `[name]-frontend${ fileSuffix }.js`;
+			},
+			uniqueName: 'webpackWcBlocksCartCheckoutFrontendJsonp',
+			library: [ 'wc', '[name]' ],
+		},
+		module: {
+			rules: [
+				{
+					test: /\.(j|t)sx?$/,
+					exclude: /node_modules/,
+					use: {
+						loader: 'babel-loader',
+						options: {
+							presets: [
+								[
+									'@wordpress/babel-preset-default',
+									{
+										modules: false,
+										targets: {
+											browsers: [
+												'extends @wordpress/browserslist-config',
+											],
+										},
+									},
+								],
+							],
+							plugins: [
+								isProduction
+									? require.resolve(
+											'babel-plugin-transform-react-remove-prop-types'
+									  )
+									: false,
+								'@babel/plugin-proposal-optional-chaining',
+								'@babel/plugin-proposal-class-properties',
+							].filter( Boolean ),
+							cacheDirectory: true,
+						},
+					},
+				},
+				{
+					test: /\.s[c|a]ss$/,
+					use: {
+						loader: 'ignore-loader',
+					},
+				},
+			],
+		},
+		optimization: {
+			concatenateModules:
+				isProduction && ! process.env.WP_BUNDLE_ANALYZER,
+			splitChunks: {
+				minSize: 200000,
+				automaticNameDelimiter: '--',
+				cacheGroups: {
+					commons: {
+						test: /[\\/]node_modules[\\/]/,
+						name: 'cart-checkout-vendors',
+						chunks: 'all',
+						enforce: true,
+					},
+					base: {
+						// A refined include blocks and settings that are shared between cart and checkout that produces the smallest possible bundle.
+						test: /assets[\\/]js[\\/](settings|previews|base|data|utils|blocks[\\/]cart-checkout-shared|icons)|packages[\\/](checkout|components)|atomic[\\/]utils/,
```

Comment:
What happens when we introduce a new folder to the mix here, which is a common process? would things break, or would bundle size increase slightly in a non-optimal way?

Reply:
@senadir nothing would break, if that was the case I would feel uncomfortable to manually declare a complex regex.

>  would bundle size increase slightly in a non-optimal way?

Yes it's possible there would be room to improve the bundle size again assuming it was a dependency that's repeated over many entry points in the bundle.